### DISCUSSION TITLE
157

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#758] [Comments]
http://jojoldu.tistory.com/157

**Comment by jojoldu** [comment]
작성자: 지나가는개발자

포스트 잘 읽었습니다. 말씀 하신 것 처럼 프론트엔드에서만 유효성 체크를 하면 문제겠지만, 프론트엔드에서 유효성 체크를 함으로써 얻을 수 있는 장점으로는 서버로 값을 보내보지 않고 값 검증을 통해 빠른 사용자경험을 제공 해 줄 수 있고, 서버가 값 검증을 해야하는 서버 리소스도 어느정도 줄여줄 수 있는 장점이 있습니다. 그래서 저는 말씀하신 첫번째 방법대로 프론트엔드와 벡엔드에서 모두 값 검증을 하는게 좋은 것 같습니다.

**Comment by jojoldu** [comment]
작성자: 개발자

보안 이슈가 있기 때문에 반드시 백엔드에서 검사를 해야 합니다.

**Comment by jojoldu** [comment]
작성자: 개발

저도 지나가는개발자님 말씀에 동의합니다. 프론트에서 유효성검사를 안하고 백엔드에서만 검사를 하면 불필요하게 많은 리소스가 서버로 유입이 될거같습니다

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

앗 ㅠ 제가 좀 오해가 있도록 제목을 작성했는것 같습니다 ㅠ
프론트엔드에서 하는게 잘못되었다는 것은 아니구요^^;;
프론트엔드에서"만" 하는 것이 "왜" 문제가 되는지에 대해 작성한 것입니다.
윗분들이 말씀하신대로 프론트엔드와 백엔드 양쪽을 다 해서 서버 리소스 줄이는 것과 UX 향상에 큰 도움이 된다는 것에 동의합니다^^
다만 저라면 백엔드에서 모든 유효성 검사를 하고, 그에 따라 "응답 형태를 프론트 엔드에서 받아서 적절하게 메세지"를 뿌리는게
많은 유효성 검사가 필요한 사이트에선 좀 더 좋지 않을까 싶습니다. (그래서 관련해서 마지막에 관련된 링크도 추가하였습니다)
위 3분다 댓글 정말 감사드립니다!

**Comment by moregorenine** [comment]
프론트 엔드에도 해야지 불필요한 리소스를 줄일수 있다는 건 동의합니다만 단점 또한 존재합니다.
위에 세분들께서 장정만 말씀하셔서 단점 언급하고자 합니다.
백엔드에는 validation 체크하기 위한 어노테이션이 제공되어 validation 체크 로직이 정말 간단하게 처리 됩니다. 하나 프론트 엔드단에서는 아직까지 백엔드에 비해 validation 체크를 지원하는 기능이 미흡하여 양쪽을 구현할 경우 백엔드 단에만 처리할 경우에 비해 2배 이상의 시간이 소요됩니다.
validation 체크 로직이 변경될 경우에도 양쪽을 함께 수정해야 하는 번거로움과 함께 운영 관리시에도 동일한 단점이 적용됩니다.
불필요한 리소스도 고려되어야 하지만 개발 기간이 정해진 프로젝트일 경우 개발공수 산정과 차후 운영관리의 효율성 또한 중요한 고려 대상이라 생각합니다.
개인적으로 validation check가 백엔드에서는 필수라고 생각하지만 프로트엔드의 validation  작성은 장점보다 단점이 크게 느껴져서 필수라고 생각하지 않네요. 있으면 좋다는건 인정하지만요.
